test/lib/buildFactory.test.js

```diff
@@ -143,6 +142,17 @@ describe('Build Factory', () => {
             });
         });
 
+        it('ignores extraneous parameters', () => {
+            const expected = {};
+            const garbage = 'garbageData';
+
+            datastore.save.yieldsAsync(null, expected);
+
+            return factory.create({ garbage, username, jobId, sha }).then(() => {
+                assert.calledWith(datastore.save, saveConfig);
```

Comment:
I kinda agree with Darren here. Class creation is treated in this test as an external run-time dependency (e.g. async), and should ultimately be tested there. One could argue that there should be a test for `base.constructor` that ensures that garbage keys are not set on the model when they are not part of the schema or otherwise expected by that class.